GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m

```diff
@@ -1002,6 +1064,47 @@ + (void)proxyAppDelegate:(id<GULApplicationDelegate>)appDelegate {
   }
 }
 
++ (void)proxySceneDelegateWithClass:(Class)sceneDelegateClass API_AVAILABLE(ios(13.0)) {
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    SEL methodSelector = @selector(scene:openURLContexts:);
+
+    Method originalMethod = class_getInstanceMethod(sceneDelegateClass, methodSelector);
+    Method swizzledMethod = class_getInstanceMethod([self class], methodSelector);
+
+    id<GULApplicationDelegate> appDelegate = [GULAppDelegateSwizzler sharedApplication].delegate;
+    NSMutableDictionary *sceneDelegateIMPDict = [objc_getAssociatedObject(appDelegate, &kGULSceneDelegateIMPDictKey) mutableCopy];
+
+    NSString *sceneDelegateClassName = NSStringFromClass(sceneDelegateClass);
+    if (sceneDelegateIMPDict[sceneDelegateClassName]) {
+      return;
+    }
+
+    IMP orignalImplementation = method_getImplementation(originalMethod);
+    NSValue *originalImplementationPointer = [NSValue valueWithPointer:orignalImplementation];
+    NSDictionary *sceneDelegateSELIMPPair = @{NSStringFromSelector(methodSelector): originalImplementationPointer};
+    sceneDelegateIMPDict[sceneDelegateClassName] = sceneDelegateSELIMPPair;
+
+    objc_setAssociatedObject(appDelegate, &kGULSceneDelegateIMPDictKey,
+                             [sceneDelegateIMPDict copy], OBJC_ASSOCIATION_RETAIN);
+
+    BOOL didAddMethod =
+    class_addMethod(sceneDelegateClass,
```

Comment:
As far as I can see, the method swizzling has been implemented here. ISA swizzling is used to proxy App Delegate which is a preferable way (method swizzling will affect any object of the class in contrast to ISA swizzling which will affect only a particular instance, so the impact on the host app will be smaller as well as risks of issues). 

I think, [GULObjectSwizzler](https://github.com/firebase/firebase-ios-sdk/blob/master/GoogleUtilities/ISASwizzler/Private/GULObjectSwizzler.h) can be used here to create a subclass for the object. Then [proxyDestinationSelector...](https://github.com/firebase/firebase-ios-sdk/blob/e88b52fb676d86df0bf665f27a899bfed46efc9c/GoogleUtilities/AppDelegateSwizzler/GULAppDelegateSwizzler.m#L575) method may be used to update the method implementation.

Feel free to reach out to me to clarify the details.

Reply:
Please see the updated pr. Seems works well. I'll add unit tests later.

Reply:
Yeah, I also would feel more comfortable with the changes having tests in place.

Reply:
I mean this pr is more of a draft. The solution is not finalized yet.

Reply:
Ah, sure, no problem